hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java

```diff
@@ -369,12 +372,13 @@ public DatanodeDeletedBlockTransactions getTransactions(
               if (checkInadequateReplica(replicas, txn)) {
                 continue;
               }
+              metrics.setNumBlockDeletionTransactionDataNodes(dnList.size());
+              metrics.incrProcessedTransaction();
```

Comment:
getTransaction() can have skip of the transaction for metrics, so this increment needs to be done in getTransaction() only if not skipped.